lib/puppet/resource/catalog.rb

```diff
@@ -86,13 +108,12 @@ def container_of(resource)
     adjacent(resource, :direction => :in)[0]
   end
 
-  def add_one_resource(resource)
-    title_key = title_key_for_ref(resource.ref)
+  def add_one_resource(resource, title_key, idx)
```

Comment:
Why was title_key added to the call signature, instead of retrieved from the resource? I don't see it needed anywhere else, it seems like the old behavior would work for add_resource_before and add_resource_after.